scipp-imaging-2D.ipynb

```diff
@@ -224,218 +162,516 @@
+    "    group_size = (324.0  / nx_target) * (324.0 / ny_target)\n",
+    "    factor = group_size\n",
+    "    rebinned_wavelength[\"sample\"] *= factor"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Note: this cell is currently a workaround until \n",
-    "# https://github.com/scipp/scipp/issues/823 is done.\n",
-    "# Then it should be possible to group the positions label directly.\n",
-    "\n",
-    "# makes the position a DataConstProxy otherwise groupby won't take it\n",
-    "stitched[\"position\"] = stitched.labels[\"position\"]\n",
-    "dm1d = sc.groupby(stitched[\"position\"], \"detector_mapping\", Dim.Row)\n",
-    "position = dm1d.mean(Dim.Spectrum)\n",
+    "def weighted_sum(variable : sc.Variable, dim_to_sum):\n",
+    "    variable = variable.copy()\n",
     "\n",
-    "# can't do stitched.labels[\"position\"] = position because Labels won't take a DataArray\n",
-    "# also can't do stitched.labels[\"position\"] = sc.Variable(position) because then sc.convert \n",
-    "# thinks stitched.labels[\"position\"] is dimensionless (as it's actually still a DataArray)\n",
+    "    variances = sc.Variable(dims=variable.dims, values=variable.variances)\n",
+    "    d = sc.Dataset()\n",
+    "    d['data'] = variable / variances\n",
+    "    d['norm'] = sc.reciprocal(variances)\n",
     "\n",
-    "pos = np.zeros((position.shape[0], 3))\n",
-    "for i, val in enumerate(position.values):\n",
-    "    pos[i, :] = val\n",
+    "    d.masks['zero_error'] = sc.equal(variances, 0.0 * variances.unit)\n",
+    "    \n",
+    "    d = sc.sum(d, dim_to_sum)\n",
+    "    sum = d['data']/d['norm']\n",
+    "    return sum\n",
     "\n",
-    "# finally add it back!\n",
-    "stitched.labels[\"position\"] = sc.Variable(position.dims, pos, unit=sc.units.m, dtype=position.dtype)"
+    "# If we are not interested in strain regions for the unloaded sample, we can just combine all spectra to improve statistics.\n",
+    "# maybe another masking before summation, with tougher threshold to exclude boderline pixels.\n",
+    "sample_final = weighted_sum(rebinned_wavelength[\"sample\"], \"spectrum\")"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "scrolled": true
-   },
    "outputs": [],
    "source": [
-    "stitched[\"normalized_wl\"] = sc.neutron.convert(stitched[\"normalized_grpd\"], Dim.Tof, Dim.Wavelength)\n",
-    "stitched"
-   ]
+    "rebinned_wavelength.coords[\"detector_mapping\"] = imaging.make_detector_groups(bank_width, bank_width,\n",
+    "                                                                              nx_target, ny_target)\n",
+    "\n",
+    "# Group detectors for the sample_elastic / sample_plastic\n",
+    "grouped = sc.groupby(rebinned_wavelength, \"detector_mapping\").sum(\"spectrum\")\n",
```

Comment:
Is this intentionally backwards from the Mantid naming? You are grouping "spectra" into "detector mappings"? I feel two other options would be a bit clearer:

- Use `'pixel'` or `'detector'` instead of `'spectrum'` everywhere. Then here: `groupby('spectrum').sum('detector')`
- Use something like `'group'` or `'spectrum_group'` here, i.e., `groupby('group').sum('spectrum')`

Reply:
No it's a product of the original Scipp notebook grouping into detector_groups using detector_mappings and the original Mantid script loading into Spectrum

I'll go through and fix up the naming

Reply:
You may need `rename_dims`:
```python
d.rename_dims({'from':'to'})
```

Reply:
Luckily we control the loader so I renamed it to detector there, however I run into a different problem

Renaming `spectrum` to `detector` breaks the unit conversion with the following error:
`Attribute dimensions must match and not exceed dimensions of data.`

And trying to rename shows there is a dim with that name already, I've double checked and replaced all the instances of `spectrum` in this repo to no avail. 
So I've reverted it to spectrum for the time being, as I can't work on it this afternoon and Owen wants to look at it